Logging/.CSProject/Log.cs

```diff
@@ -163,21 +163,24 @@ private static bool ShouldIgnoreAssembly(Assembly assembly)
                 return;
             }
 
-            Debug.Assert(!string.IsNullOrEmpty(callerDerivedTypeName));
-            Debug.Assert(!IsHandlingLog);
+            Debug.Assert(!string.IsNullOrEmpty(loggerName));
 
-            callerPath ??= UNKNOWN_CONTEXT;
-            callerName ??= UNKNOWN_CONTEXT;
+            if (IsHandlingLog)
+            {
+                throw new Exception("A log is already being handled");
+            }
+
+            CallerInfo callerInfo = new CallerInfo(
+                loggerName,
+                callerName ?? UNKNOWN_CONTEXT,
+                callerPath ?? UNKNOWN_CONTEXT,
+                (!string.IsNullOrEmpty(callerPath) ? callerPath.Split('/', '\\').Last().Split('.').First() : UNKNOWN_CONTEXT),
+                callerLine);
 
             IsHandlingLog = true;
             foreach (ILogHandler handler in s_AdditionalHandlerList)
             {
-                handler.HandleLog(level,
-                message,
-                callerDerivedTypeName,
-                callerPath,
-                callerName,
-                callerLine);
+                handler.HandleLog(level, message, callerInfo);
```

Comment:
```suggestion
                handler.HandleLog(level, message, in callerInfo);
```

Not necessary but stylistically preferred